src/lib/input/input.ts

```diff
@@ -264,13 +264,6 @@ export class MatInput implements MatFormFieldControl<any>, OnChanges, OnDestroy,
     return this._neverEmptyInputTypes.indexOf(this._type) > -1;
   }
 
-  /** Checks whether the input is invalid based on the native validation. */
-  protected _isBadInput() {
```

Comment:
can't get rid of this, it's needed for cases where the browser considers the value invalid, like this: https://jsfiddle.net/3bbhk7ka/

Reply:
Oh, good to know. I've re-added it.